mediaapi/routing/routing.go

```diff
@@ -83,26 +85,33 @@ func makeDownloadAPI(
 	activeRemoteRequests *types.ActiveRemoteRequests,
 	activeThumbnailGeneration *types.ActiveThumbnailGeneration,
 ) http.HandlerFunc {
-	return prometheus.InstrumentHandler(name, http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		req = util.RequestWithLogging(req)
+	return promhttp.InstrumentHandlerCounter(
+		promauto.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: name,
+				Help: "Total number of media_api requests for either thumbnails or full downloads",
+			},
+			[]string{"code"},
+		), http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+			req = util.RequestWithLogging(req)
 
-		// Set common headers returned regardless of the outcome of the request
-		util.SetCORSHeaders(w)
-		// Content-Type will be overridden in case of returning file data, else we respond with JSON-formatted errors
-		w.Header().Set("Content-Type", "application/json")
-
-		vars, _ := common.URLDecodeMapValues(mux.Vars(req))
-		Download(
-			w,
-			req,
-			gomatrixserverlib.ServerName(vars["serverName"]),
-			types.MediaID(vars["mediaId"]),
-			cfg,
-			db,
-			client,
-			activeRemoteRequests,
-			activeThumbnailGeneration,
-			name == "thumbnail",
-		)
-	}))
+			// Set common headers returned regardless of the outcome of the request
+			util.SetCORSHeaders(w)
+			// Content-Type will be overridden in case of returning file data, else we respond with JSON-formatted errors
+			w.Header().Set("Content-Type", "application/json")
+			vars := mux.Vars(req)
+			Download(
+				w,
+				req,
+				gomatrixserverlib.ServerName(vars["serverName"]),
+				types.MediaID(vars["mediaId"]),
+				cfg,
+				db,
+				client,
+				activeRemoteRequests,
+				activeThumbnailGeneration,
+				name == "thumbnail",
+			)
+		},
+		))
```

Comment:
The indentation doesn't look right to me here.

Reply:
![image](https://user-images.githubusercontent.com/1342360/71095054-e09d1c00-21a3-11ea-8208-7eebffd5cfaf.png)

I think it's right? It's due to a function inside of an argument.

Reply:
Anyways, refactored now.